clause.go

```diff
@@ -89,30 +89,17 @@ func (c *Clause) PreAction(action Action) *Clause {
 
 // HintAction registers a HintAction (function) for the flag to provide completions
 func (c *Clause) HintAction(action HintAction) *Clause {
-	c.addHintAction(action)
+	c.addUserCompletionWords(action()...)
```

Comment:
Sure. Sometimes Kingpin is used in long-lived applications (eg. network services) for parsing remote commands. If the hint actions are hitting database, filesystem, etc. their output can change.